assets/js/main.js

```diff
@@ -0,0 +1,308 @@
+var taskName = document.getElementById("taskName");
+var assignee = document.getElementById("assignee");
+
+var addTaskButton = document.getElementById("addTaskButton");
+var clearTaskButton = document.getElementById("clearTaskButton");
+
+var tasksData = document.getElementById("tasksData");
+var tasksDataDone = document.getElementById("tasksDataDone");
+
+var tasksArray = [];
+
+var deleteIcon = document.querySelector(".deleteIcon");
+var deleteAllButton = document.querySelector("#deleteAllButton");
+
+var currentIndex;
+
+var countTask = document.querySelector("#countTask");
+var countDone = document.querySelector("#countDone");
+
+var toggledBtn = document.querySelector("#toggledBtn");
+var toggle = 0;
+
+var nameAlert = document.getElementById("nameAlert");
+
+// to check if the local storage empty or not
+if (localStorage.getItem("tasksList") == null) {
+  tasksArray = [];
+} else {
+  tasksArray = JSON.parse(localStorage.getItem("tasksList"));
+  displayData();
+  counter();
+}
+// ----------------------------------------------------------------------------------------------------------
+// Add Task Button
+addTaskButton.onclick = function () {
+  if (addTaskButton.innerHTML == "Add-Task") {
+    addTask();
+    // sweet alert
+    Swal.fire({
+      position: "top-center",
+      icon: "success",
+      title: "The task has been added successfully",
+      showConfirmButton: false,
+      timer: 1500,
+    });
+    addTaskButton.setAttribute("disabled", "disabled");
+  } else {
+    updateTask();
+    // sweet alert
+    Swal.fire({
+      position: "top-center",
+      icon: "success",
+      title: "The task has been updated successfully",
+      showConfirmButton: false,
+      timer: 1500,
+    });
+    document.getElementById("addTaskButton").innerHTML = "Add-Task";
+  }
+
+  displayData();
+  counter();
+  clear();
+};
+// add task function
+function addTask() {
+  var done = 0;
+  var taskObject = {
+    done: done,
+    name: taskName.value,
+    assignee: assignee.value,
+  };
+  tasksArray.push(taskObject);
+  localStorage.setItem("tasksList", JSON.stringify(tasksArray));
```

Comment:
move this to its own function.
```suggestion
const setLocalStorage = (key, value) =>{ 
  localStorage.setItem(key, JSON.stringify(value));
}

setLocalStorage("tasksList", tasksArray);
```